src/osd/CMakeLists.txt

```diff
@@ -46,7 +46,7 @@ endif()
 add_library(osd STATIC ${osd_srcs})
 target_link_libraries(osd
   PUBLIC dmclock::dmclock
-  PRIVATE os heap_profiler cpu_profiler ${CMAKE_DL_LIBS})
+  PRIVATE os heap_profiler cpu_profiler win32_common ${CMAKE_DL_LIBS})
```

Comment:
where is `win32_common` target defined? this breaks the build.

Reply:
Oh, right. There was another patch from the chain that was adding it, my bad. Fixing it right away.

Reply:
so we are back to https://github.com/ceph/ceph/pull/32704#discussion_r407216945 .

Reply:
> so we are back to [#32704 (comment)](https://github.com/ceph/ceph/pull/32704#discussion_r407216945) .

What do you mean? win32_common is defined here [1] and doesn't affect linux builds:

Reply:
> so we are back to [#32704 (comment)](https://github.com/ceph/ceph/pull/32704#discussion_r407216945) . this is just confusing as it seems osd now links against win32_common..
> 
> could you add a check, something like
> 
> ```cmake
> if(WIN32)
>   target_link_libraries(osd
>     PRIVATE ceph-common)
> fi()
> ```

That was @wjwithagen suggestion [1] and IMO it was a really good one. The idea is that as opposed to Linux shared libaries, Windows DLLs can't have unresolved symbols and libosd is in fact using ceph-common functions. It was much easier to do this rather than add checks wherever ceph-common was needed (quite a few places).

[1] #31981 (comment)

Reply:
> in fact using ceph-common functions

yeah, if you could help fix the FTBFS, that'd be better.

but i don't think your reply addresses my concern regarding to "osd links against win32_common".

> It was much easier to do this rather than add checks wherever ceph-common was needed (quite a few places).

in addition to "it is easier", i also want to make sure that it's correct and it looks correct.

Reply:
My concern is that I got two reviews, asking me to do opposite things. I'm ok either way as long as this PR merges. I can add a separate check and drop that "win32_common" variable, which is a convenient way of linking against ceph-common but only when targeting Windows.

Reply:
okay, let's wait for @wjwithagen 's input then. on the other hand,

> For now, we're mostly interested in the client side, allowing Windows hosts to consume rados, rbd and cephfs resources.

if what you are after is to build the client support, why would you want to build osd on windows? AFAICT, osd is one of the daemons, i.e., it's on server side.

Reply:
Unfortunately some of the dependencies cascaded to that. I'm doing a quick build so that I can say which one exactly.

FWIW, here's a subsequent commit that was using that win32_common variable, avoiding lots of IF(WIN32) checks: https://github.com/ceph/ceph/pull/32779/commits/c0736b703e5ccbba3bb9040816dff10178c10597

Reply:
Double checked, an luckily we don't actually need libosd. I may have mistaken it with libosdc or the dependency may have been removed. I've dropped this so I can mark this as resolved.